sbi/inference/snpe/snpe_base.py

```diff
@@ -435,6 +420,26 @@ def _train(
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
 
+    def _loss(
```

Comment:
I am not sure about appending `_batch` all over the place -- our `x` and `theta` have batch dimension almost everywhere and we don't name them `_batch`, right? I do see a point in doing it in the caller, but in this function I am hesitant.

Reply:
To which I add - aren't masks also batched?